operations/CSM_product_management/Configure_Non-Compute_Nodes_with_CFS.md

```diff
@@ -182,4 +205,39 @@ Prior to running NCN personalization, gather the following information:
    using the JSON from the step 3.
 
 > **NOTE:** The CSM configuration layer _MUST_ be the first layer in the
-> NCN personalization CFS configuration.
\ No newline at end of file
+> NCN personalization CFS configuration.
+
+<a name="auto_apply_csm_config"></a>
+## Automatically Apply CSM Configuration to NCNs
+
+CSM configuration can automatically be applied to NCNs by running the
+`apply_csm_configuration.sh` script.
+
+```bash
+ncn-m001# /usr/share/doc/csm/scripts/install/configuration/apply_csm_configuration.sh
+```
+
+### Automatic CSM Configuration Steps
+ 
+By default the script will perform the following steps:
+1. Finds the latest installed release version of the CSM product stream.
+1. Finds the latest commit on the release branch of the `csm-config-management` repo.
+1. Creates or updates the `ncn-personalization.json` configuration file.
+1. Finds all nodes in HSM with the `Management` role.
+1. Disables configuration for all NCN nodes.
+1. Updates the `ncn-personalization` configuration in CFS from the `ncn-personalization.json` file.
+1. Enables configuration for all NCN nodes, and sets their desired configuration to `ncn-personalization`.
+1. Monitors CFS until all NCN nodes have successfully completed, or failed, configuration.
+
+### Automatic CSM Configuration Overrides
+
+The script also supports several flags to override these behaviors:
+- csm-release: Overrides the version of the CSM release that is used.  Defaults to the latest version.
+- git-commit: Overrides the git commit cloned for the configuration content.  Defaults to the latest
+commit on the csm-release branch.
+- git-clone-url: Overrides the source of the configuration content.  Defaults to `https://api-gw-service-nmn.local/vcs/cray/csm-config-management.git`
+- ncn-config-file: Sets a file other than `ncn-personalization.json` to be used for the configuration.
+- xnames: A comma separated list xnames to deploy to.  Defaults to all Management nodes in HSM.
```

Comment:
I think comma-separated should be hyphenated.

Reply:
Fixed